src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Handlers/RetryHandler.cs

```diff
@@ -74,17 +74,22 @@ public class RetryHandler : DelegatingHandler
                 }
                 catch (Exception ex)
                 {
-                    if (ex.IsFatal() || attempt == RetryHandler.MaxAttempts)
+                    if (ex.IsFatal() || attempt == RetryHandler.MaxAttempts - 1)
```

Comment:
Please see my comment above, regarding changing the for loop condition. If we keep it as for(int attempt = 1; attempt <= RetryHandler.MaxAttempts; ++attempt), this should just read attempt == RetryHandler.MaxAttempts